semantic-scope-graph/src/Control/Carrier/Sketch/Fresh.hs

```diff
@@ -18,80 +19,99 @@ module Control.Carrier.Sketch.Fresh
   , module Control.Effect.Sketch
   ) where
 
-import           Control.Algebra
-import           Control.Carrier.Fresh.Strict
-import           Control.Carrier.State.Strict
-import           Control.Effect.Sketch
-import           Control.Monad.IO.Class
-import           Data.Bifunctor
+import Control.Algebra
+import Control.Carrier.Fresh.Strict
+import Control.Carrier.State.Strict
+import Control.Effect.Reader
+import Control.Effect.Sketch
+import Control.Monad.IO.Class
+import Data.Bifunctor
+--import qualified Data.Map.Strict as Map
 import           Data.Module
 import           Data.Name (Name)
-import qualified Data.Name
+import qualified Data.Name as Name
 import           Data.ScopeGraph (ScopeGraph)
 import qualified Data.ScopeGraph as ScopeGraph
 import           Data.Semilattice.Lower
+import           GHC.Records
 import           Source.Span
 import qualified System.Path as Path
 
 -- | The state type used to keep track of the in-progress graph and
 -- positional/contextual information. The name "sketchbook" is meant
 -- to invoke an in-progress, concealed work, as well as the
 -- "sketching" of a graph.
-data Sketchbook address = Sketchbook
-  { sGraph        :: ScopeGraph address
-  , sCurrentScope :: address
+data Sketchbook = Sketchbook
+  { sGraph        :: ScopeGraph Name
+  , sCurrentScope :: Name
   } deriving (Eq, Show)
 
-instance Lower (Sketchbook Name) where
+instance Lower Sketchbook where
   lowerBound =
     let
-      initialGraph = ScopeGraph.insertScope n initialScope lowerBound
-      initialScope = lowerBound
-      n = Data.Name.nameI 0
+      initialGraph = ScopeGraph.insertScope n lowerBound lowerBound
+      n = Name.nameI 0
     in
       Sketchbook initialGraph n
 
-newtype SketchC address m a = SketchC (StateC (Sketchbook address) (FreshC m) a)
+newtype SketchC address m a = SketchC (StateC Sketchbook (FreshC m) a)
   deriving (Applicative, Functor, Monad, MonadIO)
 
-instance (Effect sig, Algebra sig m) => Algebra (Sketch Name :+: sig) (SketchC Name m) where
-  alg (L (Declare n _props k)) = do
-    Sketchbook old current <- SketchC (get @(Sketchbook Name))
+instance (Effect sig, Algebra sig m) => Algebra (SketchEff :+: Reader Name :+: Fresh :+: sig) (SketchC Name m) where
+  alg (L (Declare n props k)) = do
+    Sketchbook old current <- SketchC (get @Sketchbook)
     let (new, _pos) =
           ScopeGraph.declare
-          (ScopeGraph.Declaration (Data.Name.name n))
+          (ScopeGraph.Declaration n)
           (lowerBound @ModuleInfo)
-          ScopeGraph.Default
+          (relation props)
           ScopeGraph.Public
           (lowerBound @Span)
-          ScopeGraph.Identifier
-          Nothing
+          (getField @"kind" @DeclProperties props)
```

Comment:
Moving `DeclProperties` to its own module would allow you to avoid this `getField` call, I think, but I don’t object as is.